task/oracle-cloud-storage-upload/0.1/oracle-cloud-storage-upload.yaml

```diff
@@ -0,0 +1,501 @@
+            log "ERROR: Command timed out after ${timeout_duration}s: $*"
+            rm -f "$temp_output" "$temp_error"
+            return 124
+          else
+            # It was a real error, not a timeout - show the actual error
+            cat "$temp_error" >&2
+            cat "$temp_output"
+            rm -f "$temp_output" "$temp_error"
+            return $exit_code
+          fi
+        fi
+      }
+
+      # Function for logging with timestamps
+      log() {
+        echo "[$(date '+%Y-%m-%d %H:%M:%S')] $*"
+      }
+
+      log "=== Oracle Cloud Storage Upload Task ==="
+      timeout_cmd 30 oci --version || { log "ERROR: OCI CLI not found or not working"; exit 1; }
+
+      # Prepare OCI config directory
+      export OCI_CONFIG_DIR="/tmp/.oci"
+      mkdir -p "$OCI_CONFIG_DIR"
+
+      # Read fingerprint and private key
+      cp "$PRIVATE_KEY_PATH" "$OCI_CONFIG_DIR/oci_api_key.pem"
+      chmod 600 "$OCI_CONFIG_DIR/oci_api_key.pem"
+      FINGERPRINT=$(cat "$FINGERPRINT_PATH")
+
+      # Resolve credentials from params or workspace
+      TENANCY_OCID_FILE="$(workspaces.credentials.path)/tenancy_ocid"
+      USER_OCID_FILE="$(workspaces.credentials.path)/user_ocid"
+      REGION_FILE="$(workspaces.credentials.path)/region"
+      NAMESPACE_FILE="$(workspaces.credentials.path)/namespace"
+
+      if [[ -z "$TENANCY_OCID" && -f "$TENANCY_OCID_FILE" ]]; then
+        TENANCY_OCID=$(cat "$TENANCY_OCID_FILE")
+        log "Using tenancy OCID from credentials secret"
+      elif [[ -n "$TENANCY_OCID" ]]; then
+        log "Using tenancy OCID from parameter"
+      else
+        log "ERROR: tenancy OCID must be provided"
+        exit 1
+      fi
+
+      if [[ -z "$USER_OCID" && -f "$USER_OCID_FILE" ]]; then
+        USER_OCID=$(cat "$USER_OCID_FILE")
+        log "Using user OCID from credentials secret"
+      elif [[ -n "$USER_OCID" ]]; then
+        log "Using user OCID from parameter"
+      else
+        log "ERROR: user OCID must be provided"
+        exit 1
+      fi
+
+      if [[ -z "$REGION" && -f "$REGION_FILE" ]]; then
+        REGION=$(cat "$REGION_FILE")
+        log "Using region from credentials secret"
+      elif [[ -n "$REGION" ]]; then
+        log "Using region from parameter"
+      else
+        log "ERROR: region must be provided"
+        exit 1
+      fi
+
+      # Write OCI config after required values are resolved
+      cat <<EOF > "$OCI_CONFIG_DIR/config"
+      [DEFAULT]
+      tenancy=${TENANCY_OCID}
+      user=${USER_OCID}
+      fingerprint=${FINGERPRINT}
+      key_file=${OCI_CONFIG_DIR}/oci_api_key.pem
+      region=${REGION}
+      EOF
+      chmod 600 "$OCI_CONFIG_DIR/config"
+      export OCI_CLI_CONFIG_FILE="$OCI_CONFIG_DIR/config"
+
+      # Validate OCI configuration before proceeding
+      log "Validating OCI configuration..."
+      if ! timeout_cmd 30 oci iam user get --user-id "$USER_OCID" --query 'data.id' --raw-output >/dev/null 2>&1; then
```

Comment:
NIT: I would skip this. If config is invalid the upload will fail anyways, and this is an extra round-trip .
It does help with a more specific error back to users though